demos/http/http_demo_s3_upload/http_demo_s3_upload.c

```diff
@@ -0,0 +1,624 @@
+                   ( char * ) requestHeaders.pBuffer ) );
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      ( const uint8_t * ) DEMO_HTTP_UPLOAD_DATA,
+                                      DEMO_HTTP_UPLOAD_DATA_LENGTH,
+                                      &response,
+                                      0 );
+    }
+    else
+    {
+        LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
+                   serverHost, pPath ) );
+        LogInfo( ( "Response Headers:\n%.*s",
+                   ( int32_t ) response.headersLen,
+                   response.pHeaders ) );
+        LogInfo( ( "Response Status:\n%u",
+                   response.statusCode ) );
+        LogInfo( ( "Response Body:\n%.*s\n",
+                   ( int32_t ) response.bodyLen,
+                   response.pBody ) );
+    }
+    else
+    {
+        LogError( ( "An error occurred in uploading the file."
+                    "Failed to send HTTP PUT request to %s%s: Error=%s.",
+                    serverHost, pPath, HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    return httpStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example, using a pre-signed URL, resolves a S3 domain, establishes a TCP
+ * connection, validates the server's certificate using the root CA certificate
+ * defined in the config header, and then finally performs a TLS handshake with
+ * the HTTP server so that all communication is encrypted. After which, the
+ * HTTP Client library API is used to upload a file to a S3 bucket by sending a
+ * PUT request, and verify the file was uploaded using a GET request. If any
+ * request fails, an error code is returned.
+ *
+ * @note This example is single-threaded and uses statically allocated memory.
+ *
+ */
+int main( int argc,
+          char ** argv )
+{
+    /* Return value of main. */
+    int32_t returnStatus = EXIT_SUCCESS;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+    /* The length of the path within the pre-signed URL. */
+    size_t pathLen = 0;
+    /* The size of the file uploaded to S3. */
+    size_t fileSize = 0;
+
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t transportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t networkContext;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    for( ; ; )
+    {
+        LogInfo( ( "HTTPS Client Synchronous S3 upload demo using pre-signed PUT URL:\n%s", S3_PRESIGNED_PUT_URL ) );
+
+        /**************************** Parse Signed URL. ******************************/
```

Comment:
```suggestion
        /**************************** Parse Presigned URL. ******************************/
```

Reply:
Can we move this code to where it is needed in the `/*** Upload S3 Object File. ***/` section, like how you have the URL partsing for the GET url in the `/*** Verify S3 Object File Upload. ***/` section?

Reply:
Ah yes I had meant to move it when I moved `getUrlAddress `to `connectToServer` but missed it. Thanks for pointing it out.